File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function longestSubsequence($nums) {
        $n = count($nums);
        $xor = 0;
        $hasNonZero = false;

        foreach ($nums as $num) {
            $xor ^= $num;
            if ($num != 0) {
                $hasNonZero = true;
            }
        }

        if ($xor != 0) {
            return $n;
        }
        return $hasNonZero ? $n - 1 : 0;
    }
}

File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.cs
public class Solution {
    public int LongestSubsequence(int[] nums) {
        int n = nums.Length;
        int xor = 0;
        bool hasNonZero = false;

        foreach (int num in nums) {
            xor ^= num;
            if (num != 0) {
                hasNonZero = true;
            }
        }

        if (xor != 0) {
            return n;
        }
        return hasNonZero ? n - 1 : 0;
    }
}
